src/main/java/greencity/service/impl/PlaceServiceImpl.java

```diff
@@ -250,8 +251,41 @@ public Double averageRate(Long id) {
     @Override
     public List<PlaceByBoundsDto> getPlacesByFilter(FilterPlaceDto filterDto) {
         List<Place> list = placeRepo.findAll(new PlaceFilter(filterDto));
+        list = getPlacesByDistanceFromUser(filterDto, list);
         return list.stream()
             .map(place -> modelMapper.map(place, PlaceByBoundsDto.class))
             .collect(Collectors.toList());
     }
+
+    /**
+     * Method that filtering places by distance.
+     *
+     * @param filterDto - {@link FilterPlaceDto} DTO.
+     * @param placeList - {@link List} of {@link Place} that will be filtered.
+     * @return {@link List} of {@link Place} - list of filtered {@link Place}s.
+     * @author Nazar Stasyuk
+     */
+    private List<Place> getPlacesByDistanceFromUser(FilterPlaceDto filterDto, List<Place> placeList) {
+        FilterDistanceDto distanceFromUserDto = filterDto.getDistanceFromUserDto();
+        if (distanceFromUserDto != null
+            && distanceFromUserDto.getLat() != null
+            && distanceFromUserDto.getLng() != null
+            && distanceFromUserDto.getDistance() != null) {
+            placeList = placeList.stream().filter(place -> {
+                double userLatRad = Math.toRadians(distanceFromUserDto.getLat());
+                double userLngRad = Math.toRadians(distanceFromUserDto.getLng());
+                double placeLatRad = Math.toRadians(place.getLocation().getLat());
+                double placeLngRad = Math.toRadians(place.getLocation().getLng());
+
+                double distance = 6371 * Math.acos(
```

Comment:
what is "6371"?

Reply:
It is a constant from formula of Haversine. If you want to get distance in KM, you need to use 6371 constant. If you want to get in miles, replace it by 3959.